internal/utils/path.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/slsa-framework/slsa-github-generator/internal/errors"
+)
+
+// ErrInternal indicates an internal error.
+type ErrInternal struct {
+	errors.WrappableError
+}
+
+// ErrInvalidPath indicates an invalid path.
+type ErrInvalidPath struct {
+	errors.WrappableError
+}
+
+func PathIsUnderCurrentDirectory(path string) error {
```

Comment:
nit: add a short godoc to these functions.

Reply:
done. Let's have pre-submit to catch these in the future if we can

Reply:
Yeah, this should be captured under #450